packages/api/src/core/prepareParams.ts

```diff
@@ -153,9 +157,30 @@ export default async function prepareParams(operation: Operation, body?: unknown
   metadata = removeUndefinedObjects(metadata);
 
   if (!jsonSchema && (body !== undefined || metadata !== undefined)) {
-    throw new Error(
-      "You supplied metadata and/or body data for this operation but it doesn't have any documented parameters or request payloads. If you think this is an error please contact support for the API you're using."
-    );
+    let throwNoParamsError = true;
+
+    // If this operation doesn't have any parameters for us to transform to JSON Schema but they've
+    // sent us either an `Accept` or `Authorization` header (or both) we should let them do that.
+    // We should, however, only do this check for the `body` parameter as if they've sent this
+    // request both `body` and `metadata` we can reject it outright as the operation won't have any
+    // body data.
+    if (body !== undefined) {
```

Comment:
These changes fixed an edgecase I overlooked in the `accept` work in https://github.com/readmeio/api/pull/538 where if you did `api.operation({ accept: 'something' })` on an operation that had no parameters we'd throw the "this operation doesn't have any params" error.

The same case applied to this new `authorization` handling.